src/Sema.zig

```diff
@@ -13042,115 +13151,189 @@ fn resolveSwitchComptime(
+        if (empty_enum) {
+            return .void_value;
+        }
+
+        const resolved = try spa.resolveProngComptime(
+            child_block,
+            .special,
+            special.body,
+            special.capture,
+            .special_capture,
+            undefined, // case_vals may be undefined for special prongs
+            if (special.is_inline) cond_operand else .none,
+            special.has_tag_capture,
+            merges,
+        );
+        if (try sema.resolveSwitchDispatchComptime(
+            spa,
+            child_block,
+            operand_ty,
+            resolved,
+        )) |dispatched| {
+            next_dispatch_val = dispatched;
+            continue :dispatch;
+        } else return resolved;
     }
-    if (err_set) try sema.maybeErrorUnwrapComptime(child_block, special.body, cond_operand);
-    if (empty_enum) {
-        return .void_value;
-    }
+}
 
-    return spa.resolveProngComptime(
-        child_block,
-        .special,
-        special.body,
-        special.capture,
-        .special_capture,
-        undefined, // case_vals may be undefined for special prongs
-        if (special.is_inline) cond_operand else .none,
-        special.has_tag_capture,
-        merges,
-    );
+fn resolveSwitchDispatchComptime(
+    sema: *Sema,
+    spa: *SwitchProngAnalysis,
+    child_block: *Block,
+    operand_ty: Type,
+    resolved_prong: Air.Inst.Ref,
+) CompileError!?Value {
+    if (child_block.is_comptime) {
+        const break_inst = sema.comptime_break_inst;
+        const break_tag = sema.code.instructions.items(.tag)[@intFromEnum(break_inst)];
+        const break_data = sema.code.instructions.items(.data)[@intFromEnum(break_inst)];
+        const break_extra = sema.code.extraData(Zir.Inst.Break, break_data.@"break".payload_index);
+        const operand_src = LazySrcLoc.nodeOffset(break_extra.data.operand_src_node);
+        if (break_tag == .switch_continue) {
+            try sema.emitBackwardBranch(child_block, operand_src);
+            const coerced_resolved = try sema.coerce(child_block, operand_ty, resolved_prong, operand_src);
+            const operand_val: Air.Inst.Ref, const operand_ptr: Air.Inst.Ref = blk: {
+                const maybe_ptr = coerced_resolved;
+                if (spa.operand_is_ref) {
+                    const src = sema.code.instructions.items(.data)[@intFromEnum(spa.switch_block_inst)].pl_node.src();
+                    const val = try sema.analyzeLoad(child_block, src, maybe_ptr, operand_src);
+                    break :blk .{ val, maybe_ptr };
+                } else {
+                    break :blk .{ maybe_ptr, undefined };
+                }
+            };
+            spa.operand = operand_val;
+            spa.operand_ptr = operand_ptr;
+            return try sema.resolveConstDefinedValue(child_block, operand_src, coerced_resolved, undefined);
+        } else return null;
+    } else return null;
```

Comment:
About your last sentence, I think the `break` instruction must also be handled since the code below should be correct.

```zig
const b = foo: switch (a) {
    Item.a => |item| item,
    Item.e => |item| break :foo item,
    Item.c => bar: {
        break :bar 6;
    }
};
```

Reply:
Correct - you need to handle both cases, and propagate the `ComptimeBreak` up the stack if necessary.